File: auth_service/src/domain/dtos/account/account.dto.ts
import { User } from '../../entities/user.entity';

/**
 * @openapi
 * components:
 *   schemas:
 *     AccountDTO:
 *       type: object
 *       properties:
 *         userId:
 *           type: string
 *           format: uuid
 *         name:
 *           type: string
 *         email:
 *           type: string
 *           format: email
 *         emailVerified:
 *           type: boolean
 *         isEnabled:
 *           type: boolean
 *         createdAt:
 *           type: string
 *           format: date-time
 *         updatedAt:
 *           type: string
 *           format: date-time
 */
export class AccountDTO implements Partial<User> {
  userId: string;
  name: string;
  email: string;
  emailVerified: boolean;
  isEnabled: boolean;
  createdAt: Date;
  updatedAt: Date;

  constructor(data: AccountDTO) {
    this.userId = data.userId;
    this.name = data.name;
    this.email = data.email;
    this.emailVerified = data.emailVerified;
    this.isEnabled = data.isEnabled;
    this.createdAt = data.createdAt;
    this.updatedAt = data.updatedAt;
  }
}

File: post_service/src/main/java/com/engsoft/post_service/dtos/auth/AccountDTO.java
package com.engsoft.post_service.dtos.auth;

import com.google.gson.JsonObject;

public record AccountDTO(
        String userId,
        String name,
        String email,
        boolean emailVerified,
        boolean isEnabled,
        String createdAt,
        String updatedAt) {

    public static AccountDTO fromJson(JsonObject jsonObject) {
        return new AccountDTO(
            jsonObject.get("userId").getAsString(),
            jsonObject.get("name").getAsString(),
            jsonObject.get("email").getAsString(),
            jsonObject.get("emailVerified").getAsBoolean(),
            jsonObject.get("isEnabled").getAsBoolean(),
            jsonObject.get("createdAt").getAsString(),
            jsonObject.get("updatedAt").getAsString()
        );
    }
}